Library/Homebrew/download_strategy.rb

```diff
@@ -282,6 +283,14 @@ def initialize(name, resource)
   end
 
   def fetch
+    if ENV["HOMEBREW_ARTIFACT_DOMAIN"]
+      if @url.start_with?(BottleSpecification::DEFAULT_DOMAIN)
+        @url.sub!(BottleSpecification::DEFAULT_DOMAIN, ENV["HOMEBREW_ARTIFACT_DOMAIN"])
```

Comment:
@xu-cheng : Solution you have provided is really Awesome. I have tested it with bottle and non-bottle formula both.